src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -967,6 +981,7 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                     NullableWalker.AnalyzeIfNeeded(
                         _compilation,
                         methodSymbol,
+                        GetSyntaxNodesForConstructorNullableEnabledContext(methodSymbol.ContainingType),
```

Comment:
>GetSyntaxNodesForConstructorNullableEnabledContext [](start = 24, length = 50)

I think there was another place where we were actually looking at spans of initializers instead. Feels inconsistent, if that is the case.